syntax = "proto2";

package proto;

message Entity2d {
  message Vector {
    optional float x = 1; 
    optional float y = 2; 
  };

  message Animation {
    // Make sure these are sequential. My stupid code assumes they are.
    enum Type {
      kUndefined = 0;
      kIdle = 1;
      kMove = 2;
    }
    optional string animation_file = 1;
    optional Type type = 2;
  };

  optional Vector location = 1;
  repeated Animation animation = 2;
};
